# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

## What test is appropriate for this problem? Does CLT apply?

## Question 2

The **Null Hypothesis** is that having a black or white sounding name on a resume will have no effect on whether you get a call back or not.  The **Alternate hypothesis** is that there will be a bias:

$H_0\:is \: p_b = p_w\\
H_A \:is \: p_b \neq p_w$

---

$\hat{p}_b$ is the proportion of black sounding names getting a callback.

$\hat{p}_w$ is the proportion of white sounding names receiving call backs.

Standard Error:
$\sqrt{\frac{\hat{p}_b(1-\hat{p}_b)}{n_b} + \frac{\hat{p}_w(1-\hat{p}_w)}{n_w}} $

Margin of Error: 
$Z_{\alpha/2} * SE$. For a 95% confidence interval, the z-value is 1.96. 

Confidence Interval: $\hat{p}_b - \hat{p}_w \pm {Z_{\alpha/2} * SE}$


## What test is appropriate for this problem? Does CLT apply?

As long as there are more than 5 values for each random variable, we can do a two-tailed z-test.  First, we will check the value counts.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [19]:
call_b = sum(data[data.race=='b'].call)
print("Total number of callbacks for black-sounding names: {}".format(call_b))

Total number of callbacks for black-sounding names: 157.0


In [20]:
call_w = sum(data[data["race"]=='w'].call)
print("Total number of callbacks for white-sounding names: {}".format(call_w))

Total number of callbacks for black-sounding names: 235.0


In [21]:
n_b, n_w = data["race"].value_counts()
print("Total number of black-sounding names: {}".format(n_b))
print("Total number of white-sounding names: {}".format(n_w))

Total number of black-sounding names: 2435
Total number of white-sounding names: 2435


We can do Z-test for two proportions because there are 2435 values for both black and white applicants, 157 black sounding name call backs, and 235 white sounding name call backs.  All values are greater than 5.

In [7]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [14]:
data[data.race == 'b']["call"].describe()

count    2435.000000
mean        0.064476
std         0.245649
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max         1.000000
Name: call, dtype: float64

In [15]:
data[data.race == 'w']["call"].describe()

count    2435.000000
mean        0.096509
std         0.295346
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max         1.000000
Name: call, dtype: float64

In [34]:
call_total = data["call"].mean()
call_total

0.08049281686544418

In [20]:
call_b = data[data.race == 'b'].call.mean()
call_b

(call_b - call_total) / d

0.0644763857126236

In [25]:
call_w = data[data.race == 'w'].call.mean()
call_w

0.09650924056768417

In [37]:
n = len(data) / 2 # equal number of black and white applicants in the sample
Pnull = sum(data.call) / len(data)
sigma_null = np.sqrt(2 * Pnull * (1-Pnull) / n) # sigma for one group is sqrt(P * (1-P) / n)
sigma_null
Pnull

0.080492813141683772

## Question 5

While race is a significant factor, it is not the most important factor.  More important factors include...
In conclusion, race is a significant factor for qualified candidates, and a non-significant factor for a human being at random. Race is a significant factor for qualified candidates, but race alone won't make you qualified for the job.